mp4parse_capi/src/lib.rs

```diff
@@ -1049,13 +1049,15 @@ pub unsafe extern fn mp4parse_is_fragmented(parser: *mut mp4parse_parser, track_
 
     // check sample tables.
     let mut iter = tracks.iter();
-    match iter.find(|track| track.track_id == Some(track_id)) {
-        Some(track) if track.empty_sample_boxes.all_empty() => (*fragmented) = true as u8,
-        Some(_) => {},
-        None => return mp4parse_status::BAD_ARG,
-    }
-
-    mp4parse_status::OK
+    iter.find(|track| track.track_id == Some(track_id)).map_or(mp4parse_status::BAD_ARG, |track| {
+        match (&track.stsc, &track.stco, &track.stts) {
+            (&Some(ref stsc), &Some(ref stco), &Some(ref stts)) if stsc.samples.is_empty() &&
+                                                                   stco.offsets.is_empty() &&
+                                                                   stts.samples.is_empty() => (*fragmented) = true as u8,
+            _ => {},
+        };
```

Comment:
I think moving all this code to a single location is a good idea. And your implementation here is marvelously functional, but the lines are also very long. I tried several different forms, but couldn't find a better way to handle the nested Options. Maybe just break the line before the `if` guard in the match?